scripts/ligandtocore/QD_functions.py

```diff
@@ -0,0 +1,357 @@
+from scm.plams import *
+from qmflows import molkit
+from rdkit import Chem
+from rdkit.Chem import AllChem, rdMolTransforms
+import numpy as np
+import copy
+import os
+import itertools
+
+
+def read_database(ligand_folder, database_name='ligand_database.txt'):
+    """
+    open the database
+    if the database does not exist, create the database
+    """
+    # checks if database_name exists, if not creates database_name
+    if not os.path.exists(os.path.join(ligand_folder, database_name)):
+        with open(os.path.join(ligand_folder, database_name), 'w') as database:
+            database.write("{0:6} {1:19} {2:30} {3:34} {4:}".format(
+            'Index', 'Molecular_formula', 'pdb_filename', 'pdb_opt_filename', 'SMILES_string'))
+
+        database = [[],[],[],[],[]]
+
+    # else opens the database
+    else:
+        with open(os.path.join(ligand_folder, database_name), 'r') as database:
+            database = database.read().splitlines()
+        database = [line.split() for line in database if line]
+        database = list(np.transpose(database[1:]))
+        database = [list(item) for item in database]
+        database[4] = [Chem.MolFromSmiles(smiles) for smiles in database[4]]
+        database[4] = [Chem.AddHs(mol, addCoords=True) for mol in database[4]]
+        
+    return database
+
+
+def create_entry(ligand, opt):
+    """
+    create a new entry for the database
+    """
+    # compare the structures provided in ligand_list with the ones in mol_database
+    ligand_smiles = Chem.MolToSmiles(Chem.RemoveHs(molkit.to_rdmol(ligand)))
+
+    # create the new database_entry
+    a = []
+    b = molkit.from_rdmol(ligand).get_formula()
+    c = 'ligand_' + b + '.pdb'
+    if opt:
+        d = 'ligand_' + b + '.opt.pdb'
+    else:
+        d = 'ligand optimization disabled'
+    e = ligand_smiles
+    database_entry = [a, b, c, d, e]
+
+    return database_entry
+
+
+def write_database(database_entries, ligand_folder, database, database_name='ligand_database.txt'):
+    """
+    write the new database entries to the database
+    """
+    # check if any previous indices are present in database[0]  
+    if not database:
+        j = -1
+    else:
+        try:
+            j = int(database[0][-1])
+        except:
```

Comment:
You are catching all kind of error, I guess that's not the intended behavior you just one to catch the exception